src/commands/deployAzurePipeline/configurePipeline.ts

```diff
@@ -0,0 +1,50 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from 'vscode-azureextensionui';
+
+export async function configurePipeline(context: IActionContext, target: any): Promise<void | undefined> {
+    const deployToAzureExtensionInstalled = await isDeployToAzureExtensionInstalled();
+    if (!deployToAzureExtensionInstalled) {
+        return undefined;
+    }
+
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    if (!cloudExplorer.available) {
+        return undefined;
+    }
+
+    const clusterTarget = cloudExplorer.api.resolveCommandTarget(target);
+    if (clusterTarget && clusterTarget.cloudName === "Azure" && clusterTarget.nodeType === "resource" && clusterTarget.cloudResource.nodeType === "cluster") {
+        const configurePipelineCommand = 'configure-cicd-pipeline';
+        await executeDeployToAzureExtensionInstalled(configurePipelineCommand, clusterTarget);
+    } else {
+        vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+        return undefined;
+    }
+
+}
+
+async function isDeployToAzureExtensionInstalled(): Promise<boolean> {
+    const deployToAzureExtensionId = 'ms-vscode-deploy-azure.azure-deploy';
+    const pipelinesExtension = vscode.extensions.getExtension(deployToAzureExtensionId);
+    if (!pipelinesExtension) {
+        vscode.window.showWarningMessage('Please install/enable `Deploy to Azure` extension and start again.');
+        await vscode.commands.executeCommand('extension.open', deployToAzureExtensionId);
+        return false;
+    }
+    return true;
+}
+
+async function executeDeployToAzureExtensionInstalled(commandToRun: string, cluster: any): Promise<void> {
+    const listOfCommands = await vscode.commands.getCommands(true);
+    if (listOfCommands.find((commmand: string) => commmand === commandToRun)) {
+        try {
+            vscode.commands.executeCommand(commandToRun, cluster);
```

Comment:
I thought the point of putting this code into the AKS extension instead of the Deploy To Azure extension was that DTA could define a contract, and the code on the AKS side would marshal the cluster data into that format, so that if AKS changed its internal format we wouldn't break DTA.  If you're just passing the cluster tree object, then you don't get this decoupling or the safety that comes with it, and you might as well just hang the DTA command directly off the cluster tree rather than implementing and maintaining a stub command here...?

Reply:
True. Have defined a new structure which should solve the issue.